File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    answer = ''
    numbers = list(map(str, numbers))
    
    numbers.sort(key = lambda x : x*3, reverse=True)
    for i in numbers:
        answer+=i
    
    return str(int(answer))

# n^2 가능한 모든 문자열 순서로 만들기
# str -> int -> 최댓값 -> str 반환

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> v;
    for(auto n:numbers){
        v.push_back(to_string(n));
    }
    sort(v.begin(), v.end(), [](string a, string b){
        return a+b>b+a;
    });
    for(auto a:v){
        answer+=a;
    }
    if(answer[0] == '0') return "0";
    return answer;
}